### PR TITLE
Implement setHash() and getHash(). Fixes #43

### LINKED ISSUE #43: Hash support?
Does flow router support url hashes? 

I'm considering the use of hashes as a way to control the UI (i.e. show/hide modals) of a single page (single route). This is a pattern that Iron Router does a poor job of explaining but i believe is a very common pattern. Thanks

### PR DESCRIPTION
**Pull request by hpx7**

### CODE REVIEW

**Comment by arunoda** [comment]
Thanks. This is great. 
We need some test cases :)

**Comment by hpx7** [comment]
Added tests :)

**Comment by arunoda** [comment]
Great. I'll do a review today and merge these.

**Comment by hpx7** [comment]
@arunoda have you gotten a chance to check this out yet?

File: client/router.js [hunk 1]
```diff
@@ -79,11 +81,15 @@ Router.prototype.path = function(pathDef, fields, queryParams) {
...
+
   return path;
 };
 
-Router.prototype.go = function(pathDef, fields, queryParams) {
-  var path = this.path(pathDef, fields, queryParams);
+Router.prototype.go = function(pathDef, fields, queryParams, hash) {
+  var path = this.path(pathDef, fields, queryParams, hash);
 
   if (this._current.path !== path) {
```
Comment (arunoda) [inline client/router.js]: Self note: `this._current.path` does not contain the hash value. So, due to that reason, this equation will be false every time.

If we set the hash to null, theoretically it should remove. But due this reason, path and the `._current.path` is the same. So, we can't remove hash at all.

**Comment by arunoda** [comment]
I tested this and all working. But, we can't remove hash and I already know the reason.

I'm working next to add IE9 support. So, we need to use hashbang for the URL. So, setting hash will be something weird. 

So, I'm thinking to remove APIs to set hash from flow. But, we can get the hash reactively. What do you think?

**Comment by arunoda** [comment]
Hi, Thanks for the PR. 
But, hash supports seems to confuse our API. 
I think we can think about this later on but not for 2.0

So, I'm closing this for now.

**Pull request closed without merging**